### PR TITLE
Fix #5204: Support AWS Cognito SSO for user authentication

### LINKED ISSUE #5204: Support AWS Cognito SSO for user authentication
We should add support for AWS Cognito SSO for user authentication

### PR DESCRIPTION
**Pull request by vivekratnavel**
### Describe your changes :
See #5204 

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/1051198/170802852-81de4de6-a5d9-4466-9615-5e26e119702c.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by harshach** [comment]
@vivekratnavel don't we need to introduce any ingestion side configs for this?

File: openmetadata-ui/src/main/resources/ui/src/router/AppRouter.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ const AppRouter = () => {
   ) : (
     <>
       {authConfig?.provider === AuthTypes.GOOGLE ||
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/router/AppRouter.tsx:40]: @vivekratnavel , Can we do something like this for better readability.

```
const authCheck = authConfig?.provider === AuthTypes.GOOGLE ||
      authConfig?.provider === AuthTypes.AWS_COGNITO ||
      authConfig?.provider === AuthTypes.CUSTOM_OIDC

```
```suggestion
      {authCheck ? <AuthenticatedAppRouter />:(...)
```

File: openmetadata-ui/src/main/resources/ui/src/utils/UserDataUtils.ts [hunk 2]
```diff
@@ -67,9 +67,12 @@ export const getUserDataFromOidc = (
   const images = oidcUser.profile.picture
     ? getImages(oidcUser.profile.picture)
     : undefined;
+  const profileEmail = oidcUser.profile.email;
+  const email = isEmpty(profileEmail) ? userData.email : profileEmail;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/UserDataUtils.ts:71]: @vivekratnavel,  Can't this be like this?
If `profileEmail` is an empty string, the ternary will consider it as a false value, then I think we don't need the `isEmpty` method there.

```suggestion
  const email = profileEmail ? userData.email : profileEmail;
```

**Comment by vivekratnavel** [comment]
@harshach I have created #5206 to track the changes for the ingestion side.

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by harshach — APPROVED** [review]

**Pull request merged**